Form/Type/PolyCollectionType.php

```diff
@@ -130,18 +131,52 @@ function (FormView $prototype) {
             }
         }
     }
+    
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'infinite_form_polycollection';
+    }
 
     /**
      * {@inheritdoc}
      */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $this->internalConfigureOptions($resolver);
+        
+        $resolver->setAllowedTypes('types', 'array');
```

Comment:
switching the way to use the OptionsResolver based on the Form component version is wrong IMO. Symfony 2.7 will still call ``setDefaultOptions`` on form types, even if it checks for usage of the forward compatibility layer